packages/core/core.ts

```diff
@@ -124,11 +127,50 @@ export function startFuzzingAsyncNoInit(
 	return Fuzzer.startFuzzingAsync(fuzzFn, fuzzerOptions);
 }
 
-export function stopFuzzingAsync() {
+export function stopFuzzing(err: unknown, expectedErrors: string[]) {
+	// No error found, check if one is expected.
+	if (!err) {
+		if (expectedErrors.length) {
+			console.error(
+				`ERROR: Received no error, but expected one of [${expectedErrors}].`
+			);
+			Fuzzer.stopFuzzingAsync(ERROR_UNEXPECTED_CODE);
+		}
+		return;
+	}
+
+	// Error found and expected, check if it's the correct one.
+	if (expectedErrors.length) {
+		const name = errorName(err);
+		if (expectedErrors.includes(name)) {
+			console.error(`INFO: Received expected error "${name}".`);
+			Fuzzer.stopFuzzingAsync(ERROR_EXPECTED_CODE);
+		} else {
+			printError(err);
+			console.error(
+				`ERROR: Received error "${name}" is not in expected errors [${expectedErrors}].`
+			);
+			Fuzzer.stopFuzzingAsync(ERROR_UNEXPECTED_CODE);
+		}
+		return;
+	}
+
+	// Error found, but no specific one expected.
+	printError(err);
 	Fuzzer.stopFuzzingAsync();
 }
 
-export function printError(error: unknown) {
+function errorName(error: unknown): string {
+	if (error instanceof Error) {
+		return error.name;
+	} else if (typeof error === "string" || error instanceof String) {
+		return error.toString();
+	} else {
+		return "unknown";
```

Comment:
What other types could we expect here so that we'd end up returning "unknown"? Wouldn't it be better to just `return error` here as it could give more insight into what actually went wrong

Reply:
Normally `Error` objects and sometimes strings are thrown, but that's not enforced. I extended the checks to support all primitive types as well, but don't think that we actually need that for our tests. Objects or arrays can not be converted to an appropriate name to match against the expected errors, though.